tests/parser/syntax/test_for_range.py

```diff
@@ -1,7 +1,78 @@
 import pytest
 
+from pytest import raises
 from vyper import compiler
 
+fail_list = [
+    """
+@public
+def foo():
+    for i in range(10, type=int32):
+        pass
+    """,
+    """
+@public
+def foo():
+    for i in range(10, type=random):
+        pass
+    """,
+    """
+@public
+def foo():
+    x: int128 = 5
+    for i in range(10, type=uint256):
+        y: int128 = i + x
+    """,
+    """
+@public
+def foo():
+    for i in range(10, type=int128):
+        y: int128 = floor(i)
+    """
+]
+
+@pytest.mark.parametrize('bad_code', fail_list)
+def test_type_fail(bad_code):
+
+    with raises(Exception):
+        compiler.compile(bad_code)
+
+
+valid_list = [
+    """
+@public
+def foo():
+    for i in range(10, 20, type=int128):
+        pass
+    """,
+    """
+@public
+def foo():
+    x: uint256 = 5
+    for i in range(x, x + 10, type=uint256):
+        pass
+    """,
+    """
+@public
+def foo():
+    x: uint256 = 5
+    for i in range(10, type=uint256):
+        y: uint256 = i + x
+    """,
+    """
+@public
+def foo():
+    for i in range(10.0, type=decimal):
+        y: int128 = floor(i)
+    """
+]
+
+
+@pytest.mark.parametrize('good_code', valid_list)
+def test_type_success(good_code):
+    assert compiler.compile(good_code) is not None
```

Comment:
We need additional tests that confirm each of these types actually produce the output that one thinks it does. I have suspicion decimal might not as it needs `num * DECIMAL_DIVISOR // den` to work as intended.

Reply:
Do you think taking some of the tests from the types testing and combining those with the for loops will work? Or are some different tests needed?

Reply:
Have a look at the tests for the `in list()` functionality (test_for_in_list.py) for a guide on what type of tests we need here. I think creating a new test file "test_for_range." should do just fine.